test/parse/test-doc-whitespace.test.js

```diff
@@ -29,3 +29,57 @@ describe('errorHandle', () => {
 		expect({ actual, ...errors }).toMatchSnapshot()
 	})
 })
+
+describe('whitespace', () => {
+	const whitespaceToHex = (str) =>
+		str.replace(/\s/g, (c) => `#x${c.charCodeAt(0).toString(16)}`)
+	it.each([
+		[
+			'in text node before first element',
+			'\r\n<xml/>',
+			(dom) => dom.firstChild.nodeValue,
+			'#xa',
+		],
+		[
+			'in attributes',
+			'<xml attr="\r\n"/>',
+			(dom) => dom.documentElement.getAttribute('attr'),
+			'#xa',
+		],
+		[
+			'in firstChild text node',
+			'<xml>\x0D\x0A</xml>',
+			(dom) => dom.documentElement.firstChild.nodeValue,
+			'#xa',
+		],
+	])('should normalize "\\r\\n" %s', (_, xml, resolveNode, expected) => {
+		const { parser } = getTestParser()
+
+		const dom = parser.parseFromString(xml, 'text/html')
+
+		expect(whitespaceToHex(resolveNode(dom))).toBe(expected)
+	})
+	it.each([
+		[
+			'before first node',
+			'\r \n<xml/>',
+			(dom) => dom.firstChild.nodeValue,
+			'#xa#x20#xa',
+		],
+		[
+			'in firstChild text node',
+			'<xml> \r\r<child/></xml>',
+			(dom) => dom.documentElement.firstChild.nodeValue,
+			'#x20#xa#xa',
+		],
+	])(
+		'should normalize "\\r" not followed by "\\n" %s',
```

Comment:
nit: should this be tested in attribute as well?

Reply:
It can not easily, since the the algorithm in the other PR would replace it with a space.
But since we are replacing all occurrences in one go before parsing, I don't think it matters to much.